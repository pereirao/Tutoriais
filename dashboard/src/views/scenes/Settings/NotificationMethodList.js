//@flow

import React, {PureComponent} from 'react';
import {connect} from 'react-redux';
import _ from 'lodash';
import * as Ent from '@minibar/store-business/src/utils/ent';
import NotificationMethodListRow from './NotificationMethodListRow';
import NewNotificationMethodButton from './NewNotificationMethodButton';
import style_sheet from './Settings.module.css';
import MBTable from '../../elements/MBTable';
import MBLoader from '../../elements/MBLoader';
import type NotificationMethod from '../../../business/notification_method';
import {
  notification_method_actions,
  notification_method_selectors,
  notification_method_helpers
} from '../../../business/notification_method';
import I18n from '../../../localization';

const {NOTIFICATION_METHOD_COLUMN_ORDER} = notification_method_helpers;
const {allIds} = notification_method_selectors;

export class NotificationMethodList extends PureComponent {
  props: NotificationMethodListProps
  state: NotificationMethodListState

  componentWillMount() {
    this.props.fetchNotificationMethods();
  }

  renderTableHeader = (columns) => (
    <MBTable.Header>
      {columns.map(column_name => {
        if (column_name === 'mobile_summary') return null; // no header for mobile summary column
        return (
          <MBTable.HeaderCell mobile_hidden
                              key={`${column_name}header`}>{I18n.t(`ui.table.${column_name}`)}</MBTable.HeaderCell>
        );
      })}
    </MBTable.Header>
  )

  renderTableBody = (notification_methods) => {
    if (this.props.fetching) return <MBLoader/>;
    return (<MBTable.Body>{notification_methods.map(this.renderRow)}</MBTable.Body>);
  };

  renderRow = (notification_method: NotificationMethod) => (
    <NotificationMethodListRow
      columns={NOTIFICATION_METHOD_COLUMN_ORDER}
      key={notification_method.id}
      notification_method={notification_method}/>
  );

  render() {
    const {notification_methods, fetching} = this.props;
    return (
      <div className={style_sheet.tableWrapper}>
        <NewNotificationMethodButton/>
        {!_.isEmpty(notification_methods) &&
        <MBTable.Table>
          {this.renderTableHeader(NOTIFICATION_METHOD_COLUMN_ORDER)}
          {this.renderTableBody(notification_methods)}
        </MBTable.Table>
        }
        {fetching && _.isEmpty(notification_methods) && <MBLoader/>}
      </div>
    );
  }
}

const NotificationMethodListSTP = () => {
  const find_notification_methods = Ent.find('notification_method');
  return state => ({
    notification_methods: find_notification_methods(state, allIds(state)),
    fetching: notification_method_selectors.isFetching(state)
  });
};
const NotificationMethodListDTP = {
  fetchNotificationMethods: notification_method_actions.fetchNotificationMethods
};
const NotificationMethodListContainer = connect(NotificationMethodListSTP, NotificationMethodListDTP)(NotificationMethodList);

export default NotificationMethodListContainer;
