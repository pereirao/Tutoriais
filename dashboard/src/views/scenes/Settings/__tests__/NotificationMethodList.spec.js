import React from 'react';
import { NotificationMethodList } from '../NotificationMethodList';
import notification_method_factory from '../../../../business/notification_method/__tests__/notification_method.factory';


it('renders correctly', () => {
  expect(shallow(<NotificationMethodList fetchNotificationMethods={() => null} notification_methods={[notification_method_factory.build(), notification_method_factory.build()]} />)).toMatchSnapshot();
});
