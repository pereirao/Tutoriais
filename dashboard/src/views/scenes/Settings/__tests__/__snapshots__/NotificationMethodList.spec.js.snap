// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="tableWrapper"
>
  <Connect(NewNotificationMethodButton) />
  <Table>
    <Header>
      <HeaderCell
        mobile_hidden={true}
      >
        Type
      </HeaderCell>
      <HeaderCell
        mobile_hidden={true}
      >
        Value
      </HeaderCell>
      <HeaderCell
        mobile_hidden={true}
      >
        Label
      </HeaderCell>
      <HeaderCell
        mobile_hidden={true}
      >
        Status
      </HeaderCell>
      <HeaderCell
        mobile_hidden={true}
      >
        Edit
      </HeaderCell>
    </Header>
    <Body>
      <Connect(NotificationMethodListRow)
        columns={
          Array [
            "mobile_summary",
            "notification_type",
            "value",
            "label",
            "active",
            "edit",
          ]
        }
        notification_method={
          Object {
            "active": true,
            "id": 1,
            "label": "default_factory_notification_method",
            "type": "sms",
            "value": "1234567890",
          }
        }
      />
      <Connect(NotificationMethodListRow)
        columns={
          Array [
            "mobile_summary",
            "notification_type",
            "value",
            "label",
            "active",
            "edit",
          ]
        }
        notification_method={
          Object {
            "active": true,
            "id": 2,
            "label": "default_factory_notification_method",
            "type": "sms",
            "value": "1234567890",
          }
        }
      />
    </Body>
  </Table>
</div>
`;
